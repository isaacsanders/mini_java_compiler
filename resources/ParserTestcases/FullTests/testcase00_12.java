class Hello {
    public static void main (String[] args) {
        Nothing a = !new Hothing() * -null;
        
        while()
        
        if (a != 0)
           return;
        else
            a = a + 1;
    }
}

class Nothing {
    
}
