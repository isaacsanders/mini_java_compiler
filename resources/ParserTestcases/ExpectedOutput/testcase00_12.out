StmtList ::= 
Type ::= ID
CallExpr ::= NEW ID LPAREN RPAREN
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
LiteralExpr ::= NULL
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= MINUS UnaryExpr
Expr ::= UnaryExpr
Expr ::= Expr TIMES Expr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt


ERROR in line 5 and column 15: Syntax error

Error recovery to next semicolon.
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
OptExtends ::= 
ClassVarDeclList ::= 
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
Program ::= MainClassDecl ClassDeclList
